File: src/main/java/net/igneo/icv/entity/glacialImpasse/iceSpikeSpawner/IceSpikeSpawnerEntity.java
package net.igneo.icv.entity.glacialImpasse.iceSpikeSpawner;

import net.igneo.icv.entity.ICVEntity;
import net.minecraft.world.entity.EntityType;
import net.minecraft.world.entity.projectile.Projectile;
import net.minecraft.world.level.Level;

public class IceSpikeSpawnerEntity extends ICVEntity {
  private int lifetime = 0;
  
  public IceSpikeSpawnerEntity(EntityType<? extends Projectile> pEntityType, Level pLevel) {
    super(pEntityType, pLevel);
  }
  
  @Override
  public boolean isNoGravity() {
    return true;
  }
  
  @Override
  public void tick() {
    super.tick();
    
    this.setDeltaMovement(this.getOwner().getViewVector(1.0f).normalize());
    
    if (lifetime < 200) {
      lifetime++;
    } else {
      this.discard();
    }
  }
}

File: src/main/java/net/igneo/icv/entity/helmet/glacialImpasse/iceSpikeSpawner/IceSpikeSpawnerEntity.kt
package net.igneo.icv.entity.helmet.glacialImpasse.iceSpikeSpawner

import net.igneo.icv.entity.ICVEntity
import net.minecraft.world.entity.EntityType
import net.minecraft.world.entity.projectile.Projectile
import net.minecraft.world.level.Level

class IceSpikeSpawnerEntity(pEntityType: EntityType<out Projectile?>, pLevel: Level) :
    ICVEntity(pEntityType, pLevel) {
    private var lifetime = 0

    override fun isNoGravity(): Boolean {
        return true
    }

    override fun tick() {
        super.tick()
        if (this.owner == null) {
            return
        }
        this.deltaMovement = this.owner!!.getViewVector(1.0f).scale(0.3)

        if (lifetime < 200) {
            lifetime++
        } else {
            this.discard()
        }
    }
}
